### ISSUE TITLE
Document how this works with branch protection

### ISSUE THREAD

**Issue by patcon** [thiagokokada/merge-upstream#12]
This is definitely the best github action for keeping fork branches in sync like the GitHub UI allows! Thank you so much for maintaining it :))

I just wanted to offer a few suggestions, to help people like me find/understand the benefits of this Action over other ones.

- branch protection for branches that a maintainer wants to auto-sync is quite challenging. the other solutions (non-API) don't work well, as it's very complicated and hacky. Example: https://github.com/CivicTechTO/polis/pull/55

### Features of your action
- makes use of "fork sync" API endpoint, which is accounted for in existing permission toggles for branch protection
- allows using classic branch protection (no need to use new branch protection rulesets)
- allows "Fork Sync" GitHub UI to continue working (other solutions prevent this)

### Proposals
- [ ] Add mention of "fork sync" to Action's name, so that people can find it more easily
  - Helpful keywords for search are, in my mind:
    - "fork sync" (because used in GitHub UI),
    - "merge" & "fast-forward" (because this is the git commands/terms ppl might be seeking),
    - "GitHub API" (because it uses the API), and
    - "upstream merge" (because this is the what GitHub API docs call the endpoint)
- [ ] Make it clear in the documentation of features (maybe even the action name) that this works with branch protection, because it honours this special use-case in the classic branch protection config: <img width="904" alt="Screenshot 2024-10-29 at 2 14 13 PM" src="https://github.com/user-attachments/assets/a11a4e7c-f425-47db-837e-a220ec110f83">
- [ ] Document how to enable branch protection
- [ ] Include sample config for `repository-settings` app, so that ppl can understand very clearly at a glance how it works: https://github.com/repository-settings/app
- [x] ~rename repo to communicate more clearly, e.g. `merge-upstream-fork-sync-api-action`~ `wont-do`

Thanks again! If these suggestions look good to you, I'm happy to start documenting them myself in separate PRs.

**Comment by thiagokokada** [comment]
I am against renaming the action, for one the name is too long, for other this would probably be a breaking change, but anything else seems reasonable.

Thanks @patcon.

**Comment by patcon** [comment]
i respectfully disagree on both points[1][2], but fully respect your agency as project maintainer to make whichever choices you wish :)

[1] GitHub redirects all urls (web, git remotes, etc) for renamed repos in ways that don't break URI's
[2] as someone who is on v3 version of a solution (thoroughly testing/reviewing many actions in that process) and only just now found your amazing 1-star GitHub Action, I feel confident that a bigger concern is being _found and recognised_ as a candidate tool for usage, not having a name that is subjectively "too long" 🙏

**Comment by thiagokokada** [comment]
> [1] GitHub redirects all urls (web, git remotes, etc) for renamed repos in ways that don't break URI's

I can later create a new repo called `merge-upstream` (because I have short memory) and break everything.

> [2] as someone who is on v3 version of a solution (testing/reviewing many actions in that process) and only just now found your 1-star GitHub Action, I feel confident that a bigger concern is being found and recognised as a candidate tool for usage, not having a name that is subjectively "too long"

While I am happy that this helps you I created this mostly for my own use case, so I am not too concerned people are not finding this action.

**Comment by patcon** [comment]
Otherwise, will submit PRs for other aspects after I implement a solution using your Action, and have something to repurpose into a PR! Thanks!
